LiveData

JsonHrdwick opened the pull request:
Allows user to visualize and manipulate data as it is being processed by the MassSpec.

For initial commit, a temporary button is implemented in the main menu bar and specifically listens to `[user.home]\IdeaProjects\TripoliTestData\IsotopxPhoenixTIMS\KU_IGL\IsolinxVersion2\GHR1 230921 F08b.RAW\LiveData`

Manually dragging and dropping in the live data files from another directory simulates live data being created in real time. On first read, the plots window is loaded and populated. On subsequent reads, the plot window is simply updated with the new data.

Future work is needed to refine, test, and expand the functionality to be more robust and encapsulate all types of MassSpec outputs

bowring commented:
@JsonHrdwick - you can fix the merge conflicts by using the “code/reformat file” with all the checkboxes on for the file analysisManagerController and then pushing the file to your pull request.  Sorry for the inconvenience.  Cheers

JsonHrdwick commented:
@bowring Take a final look at it. Ripped out the logic for merging the analyses. Added in the mass spec context prompt that you requested. Remove livedata if not in Phoenix and removed CE tab if not in livedata

JasonHardwick-alita commented:
@bowring 
Just pushed the last of the issues we've addressed. Parameter menu is set up. Timeout is re-occurring. MassSpec choice dialog is consolidated. Live analysis is delete-able with halt choice dialog. 

Do you want the livedata analysis to get wiped from the session when the watching stops??

Other than that, I think we're good to go.

JsonHrdwick commented:
@bowring
Just pushed the last of the issues we've addressed. Parameter menu is set up. Timeout is re-occurring. MassSpec choice dialog is consolidated. Live analysis is delete-able with halt choice dialog.

Do you want the livedata analysis to get wiped from the session when the watching stops??

Other than that, I think we're good to go.
(Sorry, had used the wrong account)

bowring commented:
@JsonHrdwick -
Looking good.
1) Warning re: livedata timeout says 45 seconds when timeout is changed during timeout.  Add period at end of message.
2) Let's make Parameters menu always on
3) let's add parameter for changing/choosing mass spec type

bowring commented:
I am getting run failure at persistent state lines 111 and 140 and parameters line 41. due to overflow error = infinite loop
also tests of reports are failing due to a change in outputs - may need to update oracles
Let's make parameter for Masssspec say it is only for determining live data location.

bowring commented:
note add this to parameters.java
 private static final long serialVersionUID = 4300973848877908846L;

also line 111 in TripoliPersistentState causes an infinite loop when creating persistent state for the first time.

fix these and you should be able to fix parameters showing only once - use debugger

JsonHrdwick commented:
Added serialID, Fixed parameters reopening menu bug (I have no idea what he was trying to do there).
Parameter menu is always on now. The issue with the parameters menu being dependent on an analysis specifically was *not* fixed in this commit. I tried to tackle it but decided to just init a dummy analysis since removing this dependency would take a lot of refactoring and honestly it should be its own branch if thats something we still want to do.
